firestore/app/src/main/java/com/google/example/firestore/kotlin/SolutionBundles.kt

```diff
@@ -0,0 +1,67 @@
+package com.google.example.firestore.kotlin
+
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.Query
+import com.google.firebase.firestore.Source
+import java.io.IOException
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URL
+
+
+/**
+ * https://firebase.google.com/docs/firestore/solutions/serve-bundles
+ */
+abstract class SolutionBundles(private val db: FirebaseFirestore) {
+
+    // [START fs_bundle_load]
+    @Throws(IOException::class)
+    fun getBundleStream(urlString: String?): InputStream {
+        val url = URL(urlString)
+        val connection = url.openConnection() as HttpURLConnection
+        return connection.inputStream
```

Comment:
Does firestore buffer the stream internally? otherwise it may be very inefficient to read from an unbuffered stream

Reply:
It reads chunks into its own buffer: https://github.com/firebase/firebase-android-sdk/blob/6ce78805aeb0c87c96026a97f65e2f787ed8ab79/firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleReader.java